runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -326,8 +326,15 @@ public static DataflowRunner fromOptions(PipelineOptions options) {
     }
 
     DataflowRunnerInfo dataflowRunnerInfo = DataflowRunnerInfo.getDataflowRunnerInfo();
+    // Adding the Java version to the SDK name for user's and support convenience.
+    String javaVersion =
+        Float.parseFloat(System.getProperty("java.specification.version")) >= 9
+            ? "(JDK 11 environment)"
+            : "(JRE 8 environment)";
     String userAgent =
```

Comment:
Can you elaborate why we mix jre and jdk here?

My understanding of this article is that java.specification.version adheres to current jre specification.
https://docs.oracle.com/javase/7/docs/technotes/guides/versioning/spec/versioning2.html

Reply:
Yeah. For Java 11 there is no "JRE" - the only distribution for Java is the JDK. The containers that we provide have the Java 11 JDK.

For Java 8, there is a development distribution, and a runtime distribution (JRE). For the containers we provide the JRE.

Reply:
Thank you for clarification.